build_image.sh

```diff
@@ -13,11 +13,12 @@
 }
 
 ## Retrieval short version of Git revision hash for partition metadata
-[ -z "$(git status --untracked-files=no -s --ignore-submodules)" ] || {
-    echo "Error: There is local changes not committed to git repo. Cannot get a revision hash for partition metadata."
-    exit 1
-}
-GIT_REVISION=$(git rev-parse --short HEAD)
+if [ -z "$(git status --untracked-files=no -s --ignore-submodules)" ]; then 
+    GIT_REVISION=$(git rev-parse --short HEAD)
+else
+    echo "Warning: There is local changes not committed to git repo. Cannot get a revision hash for partition metadata."
```

Comment:
We have "--untracked-files=no" so those untracked files in submodules won't break the build.
But I do agree with you that we should not force dev to commit everything before build the code, because, 1) the change might only be a temporary one, like adding some debug printing that you would definitely remove soon. Make a commit for those change will make debug process much more complicated. 2) if you are generating a build based on local changes, it's better to have "local build" information in the image version rather than only a git hash pointing to a local commit that other people will find it nowhere.